src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1061,6 +1066,115 @@ def migrate_to_multisite(self, realm_name: str, zonegroup_name: str, zone_name:
+
+        # Create the zone and system user, then modify the zone with user credentials
+        if self.create_zone(zone_name, zonegroup_name, True, True, zone_ip_url, None, None):
+            user_details = self.create_system_user(username, zone_name)
+            if user_details:
+                keys = user_details['keys'][0]
+                self.modify_zone(zone_name, zonegroup_name, True, True, zone_ip_url, keys['access_key'], keys['secret_key'])
+
+        # Restart RGW daemons and set credentials
+        self.restart_rgw_daemons_and_set_credentials()
+
+        # Get realm tokens and import to another cluster if specified
+        realm_token_info = CephService.get_realm_tokens()
+        time.sleep(5)
+        if cluster:
+            self.import_realm_token_to_cluster(cluster, realm_name, realm_token_info)
+
+        return realm_token_info
+
+    def import_realm_token_to_cluster(self, cluster, realm_name, realm_token_info):
+        for realm_token in realm_token_info:
+            if realm_token['realm'] == realm_name:
+                realm_export_token = realm_token['token']
+                break
+        multi_cluster_config_str = str(mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        multi_cluster_config = ast.literal_eval(multi_cluster_config_str)
+        for fsid, clusters in multi_cluster_config['config'].items():
+            if fsid == cluster:
+                for cluster_info in clusters:
+                    cluster_token = cluster_info.get('token')
+                    cluster_url = cluster_info.get('url')
+                    break
+        if cluster_token:
+            headers = {
+                'Accept': 'application/vnd.ceph.api.v1.0+json',
+                'Authorization': 'Bearer ' + cluster_token,
+            }
+        else:
+            headers = {
+                'Accept': 'application/vnd.ceph.api.v1.0+json',
+                'Content-Type': 'application/json',
+            }
+
+        placement_spec = {
+            "placement": {}
+        }
+
+        payload = {
+            'realm_token': realm_export_token,
+            'zone_name': 'new_replicated_zone',
+            'port': 81,
+            'placement_spec': placement_spec
+        }
+
+        if not cluster_url.endswith('/'):
+            cluster_url = cluster_url + '/'
+
+        path = 'api/rgw/realm/import_realm_token'
+
+        try:
+            response = requests.request('POST', cluster_url + path, json=payload, headers=headers, verify=False)
+            response.raise_for_status()
+            if response.status_code == 200:
+                restart_path = 'ui-api/rgw/multisite/restart_rgw_daemons_and_set_credentials'
+                restart_gateways_response = requests.request('GET', cluster_url + restart_path, headers=headers, verify=False)
+                restart_gateways_response.raise_for_status()
+                if restart_gateways_response.status_code == 200:
+                    return True
```

Comment:
IMO, we should be reusing the multi-cluster codes to do these to avoid duplicates. can you try to reuse it from the multi-cluster code?